python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -0,0 +1,231 @@
+# Copyright (c) 2020-2022, NVIDIA CORPORATION.
```

Comment:
From a quick scan it looks like the only changes in buffer.py (aside from the move) are the removal of the `DeviceBufferLike` and `as_device_buffer_like` (which is now just `as_buffer` in utils.py) right? Just want to make sure I'm not missing anything.

Reply:
Also introduction of `_init_from_host_memory()` and clean up of `deserialize()`

Reply:
Instead of `_init_from_host_memory()`, we now have `from_device_memory()` and `from_host_memory()`